atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceTest.java

```diff
@@ -0,0 +1,435 @@
+        assertThat(trackingKvs.getOverallReadInfo()).isEqualTo(readInfo);
+        assertThat(trackingKvs.getReadInfoByTable()).containsOnlyKeys(List.of(TABLE_REF));
+    }
+
+    private void validateReadInfoForPartialRead() {
+        TransactionReadInfo readInfoForPartialReads = ImmutableTransactionReadInfo.builder()
+                .bytesRead(size)
+                .kvsCalls(1)
+                .build();
+        assertThat(trackingKvs.getOverallReadInfo()).isEqualTo(readInfoForPartialReads);
+        assertThat(trackingKvs.getReadInfoByTable()).containsOnlyKeys(List.of(TABLE_REF));
+    }
+
+    private void validateReadInfoForTableAgnosticRead(String methodName) {
+        TransactionReadInfo readInfo = ImmutableTransactionReadInfo.builder()
+                .bytesRead(size)
+                .kvsCalls(1)
+                .maximumBytesKvsCallInfo(ImmutableKvsCallReadInfo.builder()
+                        .methodName(methodName)
+                        .bytesRead(size)
+                        .build())
+                .build();
+        assertThat(trackingKvs.getOverallReadInfo()).isEqualTo(readInfo);
+        assertThat(trackingKvs.getReadInfoByTable()).isEmpty();
+    }
+
+    private Map<byte[], RowColumnRangeIterator> createRowColumnRangeIteratorByByteArrayMapOfSize() {
+        return backingRowColumnRangeMapOfSize.entrySet().stream()
+                .map(entry -> new SimpleEntry<>(
+                        entry.getKey(),
+                        new LocalRowColumnRangeIterator(
+                                entry.getValue().entrySet().iterator())))
+                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+    }
+
+    private RowColumnRangeIterator createRowColumnRangeIteratorOfSize() {
+        return new LocalRowColumnRangeIterator(
+                backingValueByCellMapOfSize.entrySet().stream().iterator());
+    }
+
+    private ClosableIterator<RowResult<Value>> createClosableValueRowResultIteratorOfSize() {
+        return ClosableIterators.wrapWithEmptyClose(
+                backingValueRowResultListOfSize.stream().iterator());
+    }
+
+    private ClosableIterator<List<CandidateCellForSweeping>>
+            createClosableCandidateCellForSweepingListIteratorOfSize() {
+        return ClosableIterators.wrapWithEmptyClose(
+                backingCandidateForSweepingTableOfSize.stream().iterator());
+    }
+
+    private ClosableIterator<RowResult<Set<Long>>> createClosableLongSetRowResultIteratorOfSize() {
+        return ClosableIterators.wrapWithEmptyClose(
+                backingLongSetRowResultListOfSize.stream().iterator());
+    }
+
+    // this is in breach with comparator invariants but is only used for equality
+    // todo(aalouane): I do not know how to do this differently
+    private static <T> Comparator<T> identityComparator() {
+        return (o1, o2) -> {
+            if (o1 == o2) {
+                return 0;
+            }
+            return 1;
+        };
+    }
+}
```

Comment:
From what I could see, using this is the only way to do what I want (i.e. check that returned iterators have the elements I want with respect to `==`, i.e. not tampered with jsut wrapped) with `assertJ`

Open to alternatives (or even checking manually)